File: nav-backend/src/main/java/com/example/nav/module/category/entity/Category.java
package com.example.nav.module.category.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@TableName("nav_category")
public class Category {

    @TableId(type = IdType.AUTO)
    private Long id;
    private String name;
    private String icon;
    private Integer sortOrder;
    private Boolean visible;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: nav-frontend/src/types/category.ts
import type { Bookmark } from './bookmark'
import type { EntityId } from './common'

export interface Category {
  id: EntityId
  name: string
  icon: string
  sortOrder: number
  visible: boolean
  bookmarkCount?: number
  createdAt?: string
  updatedAt?: string
}

export interface NavigationCategory extends Category {
  bookmarks: Bookmark[]
}

export type CategoryPayload = Omit<Category, 'id' | 'bookmarkCount' | 'createdAt' | 'updatedAt'>
